src/Monitor/Monitor/ScheduledQueryRules/SetScheduledQueryRuleCommand.cs

```diff
@@ -0,0 +1,229 @@
+        [Parameter(ParameterSetName = ByInputObject, Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = "The description for this alert")]
+        [Parameter(ParameterSetName = ByResourceId, Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = "The description for this alert")]
+        public string Description { get; set; }
+
+        //
+        // Summary:
+        //     Alert name
+        [Parameter(ParameterSetName = ByRuleName, Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "The alert name")]
+        [ResourceNameCompleter("Microsoft.insights/scheduledqueryrules", nameof(ResourceGroupName))]
+        public string RuleName { get; set; }
+
+        /// <summary>
+        /// Gets or sets the ResourceGroupName parameter of the cmdlet
+        /// </summary>
+        [Parameter(ParameterSetName = ByRuleName, Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "The resource group name")]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        //
+        // Summary:
+        //     Resource tags
+        [Parameter(ParameterSetName = ByRuleName, Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = "Resource tags")]
+        [Parameter(ParameterSetName = ByInputObject, Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = "Resource tags")]
+        [Parameter(ParameterSetName = ByResourceId, Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = "Resource tags")]
+        public IDictionary<string, string> Tag { get; set; }
+
+        //
+        // Summary:
+        //     Alert status - enabled or not, supported values - "true", "false"
+        [Parameter(ParameterSetName = ByRuleName, Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = "The azure alert state - valid values - true, false")]
+        [Parameter(ParameterSetName = ByInputObject, Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = "The azure alert state - valid values - true, false")]
+        [Parameter(ParameterSetName = ByResourceId, Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = "The azure alert state - valid values - true, false")]
+        [ValidateSet("true", "false")]
+        [PSArgumentCompleter("true", "false")]
+        public string Enabled { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "Run cmdlet in the background")]
+        public SwitchParameter AsJob { get; set; }
+        #endregion
+
+        protected override void ProcessRecordInternal()
+        {
+            ResourceIdentifier resourceIdentifier = null;
+            ScheduledQueryRuleResource requestBody = null;
+
+            // ByInputObject parameter set
+            if (this.IsParameterBound(c => c.InputObject) || this.InputObject != null)
```

Comment:
@shpimpal you can remove the check for `this.InputObject != null` since we have the `ValidateNotNull` attribute on the `-InputObject` parameter

Reply:
Had to add the check as the parameter was not being binded when invoked from mock tests.